profiling/src/profiling/mod.rs

```diff
@@ -722,6 +789,24 @@ impl Profiler {
                 sample_types.extend_from_slice(&SAMPLE_TYPES[3..5]);
                 sample_values.extend_from_slice(&values[3..5]);
             }
+
+            #[cfg(feature = "timeline")]
+            if locals.profiling_experimental_timeline_enabled {
+                sample_types.push(SAMPLE_TYPES[5]);
+                sample_values.push(values[5]);
+            }
+
+            #[cfg(feature = "timeline")]
+            if locals.profiling_experimental_timeline_enabled
+                && (samples.timeline > 0 || samples.cpu_time > 0)
+            {
+                if let Ok(now) = SystemTime::now().duration_since(UNIX_EPOCH) {
+                    labels.push(Label {
+                        key: "end_timestamp_ns",
+                        value: LabelValue::Num(now.as_nanos() as i64, Some("nanoseconds")),
+                    });
+                }
+            }
```

Comment:
We should be able to combine these two `cfg()`s into an outer if with a nested inner if.